src/executables/csql.c

```diff
@@ -3330,3 +3351,246 @@ get_host_ip (unsigned char *ip_addr)
+	  csql_new_arg.db_name = strdup (csql_arg->db_name);
+	  if (p == NULL || p[0] == '\0')
+	    {
+	      csql_new_arg.passwd = (char *) NULL;	/* to fit into db_login protocol */
+	    }
+	  else
+	    {
+	      csql_new_arg.passwd = strdup (p);
+	    }
+
+	  FREE_MEM ((char *) csql_arg->user_name);
+	  FREE_MEM ((char *) csql_arg->db_name);
+	  FREE_MEM ((char *) csql_arg->passwd);
+
+	  memcpy (csql_arg, &csql_new_arg, sizeof (CSQL_ARGUMENT));
+	  fprintf (csql_Output_fp, "CONNECTED : %s %s\n", csql_arg->user_name, csql_arg->db_name);
+	  //fprintf (csql_Output_fp, "Connected.\n");
+	}
+
+    }
+  else
+    {
+#if defined(CS_MODE)
+      memset (boot_Host_connected, 0, sizeof (boot_Host_connected));
+#endif /* CS_MODE */
+
+      /*Failed to access other host or db and then access formal db_name */
+      if (db_name == NULL)
+	{
+	  db_name = strdup (csql_arg->db_name);
+	}
+
+      if (csql_Database_connected)
+	{
+	  csql_Database_connected = false;
+	  db_end_session ();
+	  db_shutdown ();
+	}
+      er_init ("./csql.err", ER_NEVER_EXIT);
+      if (db_restart_ex (UTIL_CSQL_NAME, db_name, user_name, NULL, NULL, db_get_client_type ()) != NO_ERROR)
+	{
+	  p = getpass ((char *) csql_get_message (CSQL_PASSWD_PROMPT_TEXT));
+
+	  /* try again */
+	  if (db_restart_ex (UTIL_CSQL_NAME, db_name, user_name, p, NULL, db_get_client_type ()) != NO_ERROR)
+	    {
+	      csql_Error_code = CSQL_ERR_SQL_ERROR;
+	      csql_check_server_down ();
+
+	      fprintf (csql_Output_fp, "Warning: current CSQL session is disconnected.\n");
+	      return DO_CMD_FAILURE;
+	    }
+	  else
+	    {
+	      /*login success with password */
+
+	      csql_new_arg.user_name = strdup (user_name);
+	      csql_new_arg.db_name = strdup (db_name);
+
+	      if (p[0] == '\0')
+		{
+		  csql_new_arg.passwd = (char *) NULL;	/* to fit into db_login protocol */
+		}
+	      else
+		{
+		  csql_new_arg.passwd = strdup (p);
+		}
+
+	      FREE_MEM ((char *) csql_arg->user_name);
+	      FREE_MEM ((char *) csql_arg->db_name);
+	      FREE_MEM ((char *) csql_arg->passwd);
+
+	      memcpy (csql_arg, &csql_new_arg, sizeof (CSQL_ARGUMENT));
+
+	      if (csql_arg->sysadm && au_is_dba_group_member (Au_user))
+		{
+		  au_sysadm_disable ();
+		}
+	      csql_Database_connected = true;
+
+	      if (csql_arg->trigger_action_flag == false)
+		{
+		  db_disable_trigger ();
+		}
+
+	      fprintf (csql_Output_fp, "CONNECTED : %s %s\n", csql_arg->user_name, csql_arg->db_name);
+//            fprintf (csql_Output_fp, "Connected.\n");
+	    }
+	}
+      else
+	{
+	  /*login success without password */
+	  csql_new_arg.user_name = strdup (user_name);
+	  csql_new_arg.db_name = strdup (db_name);
+	  csql_new_arg.passwd = (char *) NULL;
+
+	  FREE_MEM ((char *) csql_arg->user_name);
+	  FREE_MEM ((char *) csql_arg->db_name);
+	  FREE_MEM ((char *) csql_arg->passwd);
+
+	  memcpy (csql_arg, &csql_new_arg, sizeof (CSQL_ARGUMENT));
+
+	  if (csql_arg->sysadm && au_is_dba_group_member (Au_user))
```

Comment:
After connecting with the --sysadm option, is there no need to au_enable() when connecting to another DB as a general user?
I'm curious if there is a scenario like this.

Reply:
Thank you for your comment.
This is copy from legacy code. So, I do not understand how it works in --sysadm option.

But for a few hours ago, our CTO and we decided to not to allow --sysadm in the CONNECT session command.